src/core/memory.cc

```diff
@@ -869,13 +948,13 @@ small_pool::alloc_site_holder(void* ptr) {
 #endif
 
 void*
-cpu_pages::allocate_small(unsigned size) {
+cpu_pages::allocate_small(unsigned size, size_t sample_weight) {
```

Comment:
Sorry yeah that's not really clean right now. Something I will clean up.

I started with the bool, then when introducing the proper sampler switched to the weight and then again realised that we can't actually use the weight right now because we would have to store it so ended up using it like a bool.